### PR TITLE
Create CNAME

### PR DESCRIPTION
**Pull request by jpienaar**
Add CNAME for serving docs from iree.dev.

### CODE REVIEW

File: CNAME [hunk 1]
```diff
@@ -0,0 +1 @@
+iree.dev
```
Comment (ScottTodd) [inline CNAME:1]: Can you add a description to this PR with some context for this change?

**Review by mattwalsh — APPROVED** [review]

File: CNAME (hunk 1 above)
Comment (ScottTodd) [inline CNAME:1]: Can you test how `mkdocs gh-deploy` interacts with this file? I'm wondering if that will overwrite the `gh-pages` branch and if we'll need to add this file to https://github.com/iree-org/iree/tree/main/docs/website/overrides instead (on the `main` branch).

https://github.com/iree-org/iree/tree/main/docs/website#how-to-edit-this-documentation

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: CNAME (hunk 1 above)
Comment (ScottTodd) [reply CNAME:1]: In particular, are you planning on fully renaming https://iree-org.github.io/iree/ to docs.iree.dev or iree.dev? We'll have a bunch of links to update in that case.

Comment (jpienaar) [reply CNAME:1]: I think both of these should still work: initially iree.dev and iree-org.github.io/iree/ should match with redirect from docs.iree.dev to iree.dev (github has single domain restriction, I could also flip it and have docs.iree.dev here and iree.dev be the redirect). I think full renaming would be nice though.

Comment (jpienaar) [reply CNAME:1]: https://www.mkdocs.org/user-guide/deploying-your-docs/#custom-domains that seems likely

Comment (ScottTodd) [reply CNAME:1]: iree.dev as the base, with docs.iree.dev redirecting there SGTM (at least until we have more subdomains and find something more fitting to sit at the unqualified root name)

**Pull request closed without merging**